interface/src/lib.rs

```diff
@@ -97,15 +98,133 @@ impl std::fmt::Display for NetworkInRequest {
     }
 }
 
+#[derive(CandidType, Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Txid([u8; 32]);
+
+impl AsRef<[u8]> for Txid {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl serde::Serialize for Txid {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        serializer.serialize_bytes(&self.0)
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for Txid {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        struct TxidVisitor;
+
+        impl<'de> serde::de::Visitor<'de> for TxidVisitor {
+            type Value = Txid;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a 32-byte array")
+            }
+
+            fn visit_bytes<E>(self, value: &[u8]) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                match TryInto::<[u8; 32]>::try_into(value) {
+                    Ok(txid) => Ok(Txid(txid)),
+                    Err(_) => Err(E::invalid_length(value.len(), &self)),
+                }
+            }
+        }
+
+        deserializer.deserialize_bytes(TxidVisitor)
+    }
+}
+
+impl fmt::Display for Txid {
+    fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // In Bitcoin, you display hash bytes in reverse order.
+        //
+        // > Due to historical accident, the tx and block hashes that bitcoin core
+        // > uses are byte-reversed. I’m not entirely sure why. Maybe something
+        // > like using openssl bignum to store hashes or something like that,
+        // > then printing them as a number.
+        // > -- Wladimir van der Laan
```

Comment:
Lol 🤣

Reply:
It's a hobby project. What did you expect?   :slightly_smiling_face: